### PR TITLE
Spider QoL

### LINKED ISSUE #12364: Spider queens will get stuck on things they can't break or move around when trying to break bulbs, and ignore everything else to do so

### LINKED ISSUE #31040: spider queens have a weird auto-aim
(WEB REPORT BY: gallywix4 REMOTE: 206.221.180.138:7777)
# Revision
33db6af663121ec6f0245f40210cfee0890f95ee

# Description
the ranged attack of the spider queen has a weird auto-aim that targets people both alive and dead
# Steps to Reproduce
click in a tile with no space man/ click on space man when there is another space man closer
# What you Expected
hitting the space man i clicked on
# What Actually Happened
free cs-go hacks

### LINKED ISSUE #32084: Midround Spiders Can't Smash Windows
(WEB REPORT BY: spacedgod REMOTE: 206.221.180.138:7777)
# Revision
21738eaef1c8a1b8b1a08986b8929e6b91e293d1

# Description
player controlled spiders can't smash all windows. For no good reason, normally reachable windows can't be smashed by PC giant spiders. It seems to be a window orientation issue but fuck if I know
# Steps to Reproduce

# What you Expected

# What Actually Happened

### LINKED ISSUE #32170: Delicious spider hardcoding
so spiders are hardcoded to not be able to break windows to space, which is nice, right?
```
/mob/living/simple_animal/hostile/giant_spider/proc/performPressureCheck(var/turf/curturf)
	if(!istype(curturf))
		return 0
	var/datum/gas_mixture/myenv=curturf.return_air()
	var/pressure=myenv.return_pressure()

	for(var/checkdir in cardinal)
		var/turf/T = get_step(curturf, checkdir)
		if(T && istype(T))
			var/datum/gas_mixture/environment = T.return_air()
			var/pdiff = abs(pressure - environment.return_pressure())
			if(pdiff > SPIDER_MAX_PRESSURE_DIFF)
				return pdiff
	return 0
```
however, this also applies to player spiders, so its easy to get stuck in places like xenobio or viro if the doors are bolted unless you, get this, go out the back through the doors, breaching anyways.
This should really (somehow) be updated to not apply to player spiders. maybe make npc spiders a deepepr subtype with this set of code so it only applies to them, but that'd have a lot of copycode, maybe a var? Idunno I don't get paid.

### PR DESCRIPTION
**Pull request by adacovsk**
[qol][tweak][bugfix]

I made spider changes with this PR:
- Player controlled spiders can break glass from space and now no longer need to enter/exit through airlocks
- Added slight regen
- Potentially fixes the sticky ball. The issue was that it was using `isliving(A)` which refers to the mob type, rather than the status (`.stat`) which is `CONSCIOUS`, `UNCONSCIOUS`, or `DEAD`

Closes #32084, Closes #32170, Closes #31040, Closes #12364 as #29429 fixed it
:cl:
 * tweak: Human-controlled spiders can break windows from space and have minor regen

### CODE REVIEW

**Comment by Servalfang** [comment]
add some amount of passive health regen to player controlled spiders or maybe just spiders full stop

**Comment by gurfan** [comment]
👍 to breaking windows 👎 to making them immune to zas, fuck you deal with it

Getting pushed by airflow discourages spiders to just smash windows/firelocks indiscriminately and it gives people more ways to deal with them. As annoying as getting blown out an airlock repeatedly might be, there are other ways to get back on station if you're trapped outside.

**Comment by Eneocho** [comment]
Since you're working on spider QoL, can you PLEASE fix #31040/#32170? That fucking hardcoded autoaim shit has gotten me killed more than once.
Also add #32084 to the OP since you removed the window issue for player spooders.

**Comment by WindowsErrors** [comment]
its kinda weird how all the spiders have almost religiously avoided smashing these windows leading to space on the space station but all the external airlocks leading out to space have all been forced and stuck open

**Comment by pedr69** [comment]
>but all the external airlocks leading out to space have all been forced and stuck open

don't forget all the firelocks that closed trying to seal the breach that were smashed open and sucked into the void

**Comment by xeroxemnas** [comment]
Do player controlled spiders get spawned in or do you have to use the mind machine as a xenobio crazy person.

**Comment by adacovsk** [comment]
> Do player controlled spiders get spawned in or do you have to use the mind machine as a xenobio crazy person.

There is an antag datum called, "Spider Infestation"

**Comment by Kurfursten** [comment]
> its kinda weird how all the spiders have almost religiously avoided smashing these windows leading to space on the space station but all the external airlocks leading out to space have all been forced and stuck open

Until my work with Optimism on his new vault, mobs only had 2 options: either they could target doors, or they couldn't. If they could, this let them smash open doors but they would prioritize killing a door equal to killing a player, or a monkey, or a light. That is to say, their crusade against airlocks is because airlocks are literally on their aggro list.

The new smart flag allows mobs to smash open doors only in a pursuit, but the system could even be refined further.

File: code/modules/mob/living/simple_animal/hostile/giant_spider/base_spider.dm [hunk 1]
```diff
@@ -154,6 +157,8 @@
 /mob/living/simple_animal/hostile/giant_spider/Life()
 	if(timestopped)
 		return 0 //under effects of time magick
+	if(health > 0)
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/hostile/giant_spider/base_spider.dm]: add a check to see if health is full; this saves operations when unnecessary

File: code/modules/mob/living/simple_animal/hostile/giant_spider/base_spider.dm [hunk 2]
```diff
@@ -157,7 +157,9 @@
 /mob/living/simple_animal/hostile/giant_spider/Life()
 	if(timestopped)
 		return 0 //under effects of time magick
-	if(health > 0)
+	if (health == maxHealth)
+		return
```
Comment (Kurfursten) [inline code/modules/mob/living/simple_animal/hostile/giant_spider/base_spider.dm]: This prevents the parent from being called

return ..() would be okay

**Pull request merged**